localstack-core/localstack/services/lambda_/event_source_mapping/pollers/kinesis_poller.py

```diff
@@ -101,5 +109,9 @@ def transform_into_events(self, records: list[dict], shard_id) -> list[dict]:
                 "data": to_str(base64.b64encode(record["Data"])),
                 "approximateArrivalTimestamp": record["ApproximateArrivalTimestamp"].timestamp(),
             }
+            if self.kinesis_namespace:
+                event["kinesis"] = kinesis_fields
+            else:
+                event.update(kinesis_fields)
```

Comment:
I'm fine with this, but if we need more of this change of formatting depending on ESM/pipes, it might make sense to delegate this to another class.

Reply:
So far, it's only:
* `KinesisPoller.kinesis_namespace` (this one)
* `LambdaSender.payload_dict`

I didn't want to over-complicate the design for two special cases, but will keep an eye out if more such differences appear 👀 .